docs/api-reference.md

```diff
@@ -3,25 +3,623 @@ id: api-reference
+
+##### Example
+
+Use the following command to get the current [shopping-list](https://github.com/reimagined/resolve/tree/master/examples/shopping-list) example application's state.
+
+```sh
+curl -g -X GET "http://localhost:3000/api/query/Default/shoppingLists"
+```
+
+#### Command API
+
+You can send a command from the client side as a POST request to the following URL:
+
+```
+http://{host}:{port}/api/commands
+```
+
+The request body should have the `application/json` content type and contain a JSON representation of the command:
+
+```
+{
+  "aggregateName": aggregateName,
+  "type": commandType,
+  "aggregateId": aggregateID,
+  "payload": {
+    "param1": value1,
+    "param2": value2,
+    ...
+    "paramN": valueN
+  }
+}
+```
+
+| Name              | Type   | Description                                           |
+| ----------------- | ------ | ----------------------------------------------------- |
+| **aggregateId**   | string | The ID of an aggregate that should handle the command |
+| **aggregateName** | string | The aggregate's name as defined in **config.app.js**  |
+| **commandType**   | string | The command type that the aggregate can handle        |
+| **payload**       | object | The parameters that the command accepts               |
+
+##### Example
+
+Use the following command to add an item to the **shopping-list** example:
+
+```sh
+$ curl -X POST "http://localhost:3000/api/commands"
 --header "Content-Type: application/json" \
 --data '
 {
-    "aggregateName": "ShoppingList",
-    "aggregateId": "12345-new-shopping-list",
-    "type": "createShoppingList",
-    "payload": {
-        "name": "List 1"
-    }
+  "aggregateName":"Todo",
+  "type":"createItem",
+  "aggregateId":"root-id",
+  "payload": {
+    "id":`date +%s`,
+    "text":"Learn reSolve API"
+  }
 }
 '
 ```
+
+### Resolve-Redux Library
+
+The reSolve framework includes the client **resolve-redux** library used to connect a client React + Redux app to a reSolve-powered backend. This library provides the following HOCs:
+
+| Function Name                                     | Description                                                                                                      |
+| ------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------- |
+| [connectViewModel](#connectviewmodel)             | Connects a React component to a reSolve View Model.                                                              |
+| [connectReadModel](#connectreadmodel)             | Connects a React component to a reSolve Read Model.                                                              |
+| [connectRootBasedUrls](#connectrootbasedurls)     | Fixes URLs passed to the specified props so that they take into respect the correct root folder path.            |
```

Comment:
"take into respect" Where did you find this phrase?

Reply:
I meant to write 'into account', fixed